.github/workflows/main.yml

```diff
@@ -173,6 +174,32 @@ jobs:
             OS-${{ runner.os }},
             VM-${{ matrix.platform }},
             Py-${{ steps.python-install.outputs.python-version }}
+            
+  check_changed_folders:
+    name: Check files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get changed files in the _distutils folder
+        id: changed-files-specific-distutils
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_distutils/**
+      - name: Get changed files in the _vendor folder
+        id: changed-files-specific-vendor
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_vendor/**
+      - name: Run step if any file(s) in the docs folder change
+        if: steps.changed-files-specific-distutils.outputs.any_changed == 'true' || steps.changed-files-specific-vendor.outputs.any_changed == 'true'
+        run: |
+          echo "One or more files in the setuptools/_distutils or setuptools/_vendor folder has changed."
```

Comment:
Maybe make this into two steps to show more precise information?

Also, I'd pipe it into `tee` writing that to the GH job summary too.